plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2580,7 +2588,7 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
         checkArgument(handle.getAnalyzePartitionValues().isEmpty() || constraint.getSummary().isAll(), "Analyze should not have a constraint");
 
         HivePartitionResult partitionResult = partitionManager.getPartitions(metastore, new HiveIdentity(session), handle, constraint);
-        HiveTableHandle newHandle = partitionManager.applyPartitionResult(handle, partitionResult, constraint.getPredicateColumns());
+        HiveTableHandle newHandle = partitionManager.applyPartitionResult(handle, partitionResult, constraint.getPredicateColumns(), constraint.predicate().isPresent());
 
         if (handle.getPartitions().equals(newHandle.getPartitions()) &&
```

Comment:
Does this comparison still work? Two handles could theoretically contain the same partition information but one is loaded and the other is not?

Reply:
bump